### PR TITLE
[refactor]: #1687 bump dependencies

### LINKED ISSUE #1687: Bump dependencies
Some dependencies got stale, bump their versions. Use `cargo update` and `cargo outdated -wR`

### PR DESCRIPTION
**Pull request by mversic**
Signed-off-by: Marin Veršić <marin.versic101@gmail.com>

### Issue

Closes #1687

### CODE REVIEW

**Review by appetrosyan — APPROVED** [review]

File: core/Cargo.toml [hunk 1]
```diff
@@ -44,7 +44,7 @@ dashmap = { version = "4.0" }
 eyre = "0.6.5"
 futures = { version = "0.3.17", default-features = false, features = ["std", "async-await"] }
 parity-scale-codec = { version = "2", features = ["derive"] }
-rand = "0.7.3"
+rand = "0.8"
```
Comment (s8sato) [inline core/Cargo.toml:47]: The dependency graph has both of following 2:

* `rand v0.7.3` (a dependency of `ursa v0.3.7`)
* `rand v0.8.4`

Is it okay?

Comment (mversic) [reply core/Cargo.toml:47]: shouldn't be an issue, this happens all the time

Comment (s8sato) [reply core/Cargo.toml:47]: I found no reason to keep rand v0.7.3, but I thought there might be some reason why it has been kept so far. What do you think @eadventurous ?

Comment (mversic) [reply core/Cargo.toml:47]: in `futures/Cargo.toml` we already used `rand = 0.8`

Comment (eadventurous) [reply core/Cargo.toml:47]: It's ok to switch it I think

**Review by eadventurous — APPROVED** [review]

**Pull request merged**